### PR TITLE
DRYD-1331: NAGPRA Inventory

### PR DESCRIPTION
**Pull request by mikejritter**
**What does this do?**
* Create nagprainventory procedure 

**Why are we doing this? (with JIRA link)**
Jira: https://collectionspace.atlassian.net/browse/DRYD-1331

**How should this be tested? Do these changes have associated tests?**
* Run `npm run test` and `npm run lint` as a sanity check
* Build collectionspace with the application pr for nagpra inventory and all tenants needed for running the integration tests
* Start collectionspace 
* Run the devserver
* Run basic operations for a procedure, e.g. 
  * Create a nagpra inventory
  * Search for a nagpra inventory (checking that advanced field search terms are consistent)
  * Relate nagpra inventories   

**Dependencies for merging? Releasing to production?**
None

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
@mikejritter tested locally

### CODE REVIEW

File: src/plugins/recordTypes/nagprainventory/fields.js [hunk 1]
```diff
@@ -0,0 +1,775 @@
...
+                }),
+                view: {
+                  type: TermPickerInput,
+                  props: {
+                    source: 'nagpranotice',
+                  },
+                },
+              },
+            },
+            determinedBy: {
```
Comment (mikejritter) [inline src/plugins/recordTypes/nagprainventory/fields.js:329]: This needs to be updated to be repeatable to be in line with the spec. Likely also needs to be fixed in duty of care.

File: src/plugins/recordTypes/nagprainventory/fields.js [hunk 2]
```diff
@@ -0,0 +1,789 @@
...
+          [config]: {
+            cloneable: false,
+            messages: defineMessages({
+              inUse: {
+                id: 'field.nagprainventories_common.inventoryNumber.inUse',
+                defaultMessage: 'The identification number {value} is in use by another record.',
+              },
+              name: {
+                id: 'field.nagprainventories_common.inventoryNumber.name',
+                defaultMessage: 'Inventory/Documentation Number',
```
Comment (ray-lee) [inline src/plugins/recordTypes/nagprainventory/fields.js:56]: Need sentence case on this, so 'Inventory/documentation number'.

File: src/plugins/recordTypes/nagprainventory/fields.js [hunk 3]
```diff
@@ -0,0 +1,789 @@
...
+            },
+          },
+        },
+        originationDate: {
+          [config]: {
+            dataType: DATA_TYPE_DATE,
+            messages: defineMessages({
+              name: {
+                id: 'field.nagprainventories_common.originationDate.name',
+                defaultMessage: 'Origination Date',
```
Comment (ray-lee) [inline src/plugins/recordTypes/nagprainventory/fields.js:82]: Sentence case please.

**Review by ray-lee — CHANGES_REQUESTED** [review]

**Review by ray-lee — APPROVED** [review]
Looks good.

**Pull request merged**